eth/typing.py

```diff
@@ -32,4 +34,8 @@
     List[Tuple[Address, Dict[str, Union[int, bytes, Dict[int, int]]]]]
 ]
 
+TransactionType = Dict[str, Union[bytes, int, str]]
```

Comment:
Did you try locking this down to a `TypedDict`? Also, in general I prefer not to use the suffix `Type` for any new types / aliases that we create. It feels redundant to me (we also say `Chain`, `BaseComputation`, `BaseVM` and not `ChainType`, `BaseComputationType` or `BaseVMType`.

Reply:
As a name suggestion, maybe just `TransactionDict`?

Reply:
```
def normalize_int(value: Union[int, bytes, str, ...]) -> int
    """
    Robust to integer conversion, handling hex values, string representations,
    and special cases like `0x`.
    """
    if is_integer(value):
        return value
   ...
   ...
```

@cburgdorf @pipermerriam if the type hinting was changed as above, then the `return value` would return type ` Union[int, bytes, str, ...]` instead of `int` which would throw an error. I have tested those and tried tightening the types prior to this PR.

Reply:
Inside the `if is_integer` block, it seems reasonable to `cast(int, value)`.

Reply:
If we need to cast it, then we would need to cast it to a new variable. How do we `overwrite` the type of a variable, which is already type hinted? This is because, mypy is throwing error when trying to overwrite the type hinting of the variables which are already type hinted.
```
value = cast(int, value)
return value
```
The following error occurs `Incompatible return value type (got "Union[int, bytes, ...]", expected "int")`

Reply:
I think it's fine to: `return cast(int, value)`

I'm not sure why mypy doesn't like the cast. @cburgdorf is our resident mypy wrangler.